script/gen/cxx14

```diff
@@ -0,0 +1,490 @@
+#!/usr/bin/env python
+
+""" Extracts MK FFI's specifications from the
+    `include/measurement_kit/README.md` file and generates the corresponding
+    C++11 wrappers into the `include/measurement_kit/cxx14.hpp` file. """
+
+from __future__ import print_function
+
+import json
+import re
+import sys
+
+if sys.version_info[0] == 3:
+    STR = str
+elif sys.version_info[0] == 2:
+    STR = unicode
+else:
+    raise RuntimeError("unexpected major Python version")
+
+def event_name_to_camel_case(name):
+    """ Convert event name to camel case """
+    array = name.replace(".", "_").split("_")
+    array = [elem.capitalize() for elem in array]
+    return "".join(array)
+
+def event_name_to_snake_case(name):
+    """ Convert event name to snake case """
+    return name.replace(".", "_")
+
+def option_name_to_snake_case(name):
+    """ Convert option name to snake case """
+    return name.replace("/", "_")
+
+def gather_events_info():
+    """ Gathers info on events """
+    events = {}
+    with open("include/measurement_kit/README.md", "r") as filep:
+        current = []
+        inside = False
+        for line in filep:
+            line = line.rstrip()
+            if line == "```JSON":
```

Comment:
I would consider having a custom language descriptor (!= `JSON`) to indicate event_info JSON documents.

You may want in the future to enrich the Readme.md with other JSON formatted data and will be sad you assigned it to this (or at least add a note to the Readme.md saying that this is a strict assumption).

Alternatively maybe you can put the JSON documents for the event definitions in another place.

Reply:
Yeah, this is a solid concern. I am not sure I want to add other kind of JSON formatted data to the README, since that is meant to document this kind of JSONs. I hear your concern loud and clear, tho, and I think that saying explicitly that the documentation is used to generate code is a more robust thing to do. (I am a bit uncomfortable with not using JSON in the use case, since when you run the code you seen JSONs, so showing you what you see is helpful.)

Reply:
(I am also not comfortable with moving the JSONs to another place, since that is the place where one needs to read to understand the JSONs, basically. So having links to outside stuff will probably strike me as something that would make people nervous: too many indirections)